hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/recovery/NMMemoryStateStoreService.java

```diff
@@ -168,6 +168,7 @@ public synchronized void storeContainer(ContainerId containerId,
       int version, long startTime, StartContainerRequest startRequest) {
     RecoveredContainerState rcs = new RecoveredContainerState(containerId);
     rcs.startRequest = startRequest;
+    rcs.status = RecoveredContainerStatus.REQUESTED;
```

Comment:
Where does this come from?

Reply:
When a container is recovered, it is starting with rcs.status as null with NMMemoryStateStoreService. With leveldb, it is recovering in Requested status and then based on what keys are present is getting updated. 

  private RecoveredContainerState loadContainerState(LeveldbIterator iter,
       String keyPrefix) throws IOException {
    ContainerId containerId = ContainerId.fromString(
        keyPrefix.substring(CONTAINERS_KEY_PREFIX.length(),
            keyPrefix.length()-1));
    RecoveredContainerState rcs = new RecoveredContainerState(containerId);
    **rcs.status = RecoveredContainerStatus.REQUESTED;**

I think, for NMMemoryStateStore also, on first time loading container, it should set status as Requested.